frontend/src/components/features/settings/api-keys-manager.tsx

```diff
@@ -0,0 +1,318 @@
+      // Ensure keys is always an array
+      setApiKeys(Array.isArray(keys) ? keys : []);
+    } catch (error) {
+      displayErrorToast(
+        retrieveAxiosErrorMessage(error as AxiosError) ||
+          t(I18nKey.ERROR$GENERIC),
+      );
+      // Set empty array on error
+      setApiKeys([]);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchApiKeys();
+  }, []);
+
+  const handleCreateKey = async () => {
+    if (!newKeyName.trim()) {
+      displayErrorToast(t(I18nKey.ERROR$REQUIRED_FIELD));
+      return;
+    }
+
+    try {
+      setIsCreating(true);
+      const newKey = await ApiKeysClient.createApiKey(newKeyName);
+
+      setNewlyCreatedKey(newKey);
+      setCreateModalOpen(false);
+      setShowNewKeyModal(true);
+      await fetchApiKeys();
+      displaySuccessToast(t(I18nKey.SETTINGS$API_KEY_CREATED));
+    } catch (error) {
+      displayErrorToast(
+        retrieveAxiosErrorMessage(error as AxiosError) ||
+          t(I18nKey.ERROR$GENERIC),
+      );
+    } finally {
+      setIsCreating(false);
+      setNewKeyName("");
+    }
+  };
+
+  const handleDeleteKey = async () => {
+    if (!keyToDelete) return;
+
+    try {
+      setIsDeleting(true);
+      await ApiKeysClient.deleteApiKey(keyToDelete.id);
+      await fetchApiKeys();
+      setDeleteModalOpen(false);
+      setKeyToDelete(null);
+      displaySuccessToast(t(I18nKey.SETTINGS$API_KEY_DELETED));
+    } catch (error) {
+      displayErrorToast(
+        retrieveAxiosErrorMessage(error as AxiosError) ||
+          t(I18nKey.ERROR$GENERIC),
+      );
+    } finally {
+      setIsDeleting(false);
+    }
+  };
+
+  const formatDate = (dateString: string | null) => {
+    if (!dateString) return "Never";
+    return new Date(dateString).toLocaleString();
+  };
+
+  return (
+    <>
+      <div className="flex flex-col gap-6">
+        <div className="flex items-center justify-between">
+          <BrandButton
+            type="button"
+            variant="primary"
+            onClick={() => setCreateModalOpen(true)}
+          >
+            {t(I18nKey.SETTINGS$CREATE_API_KEY)}
+          </BrandButton>
+        </div>
+
+        <p className="text-sm text-gray-300">
+          {t(I18nKey.SETTINGS$API_KEYS_DESCRIPTION)}
+        </p>
+
+        {isLoading && (
+          <div className="flex justify-center p-4">
+            <LoadingSpinner size="large" />
+          </div>
+        )}
+        {!isLoading && Array.isArray(apiKeys) && apiKeys.length > 0 && (
+          <div className="border border-tertiary rounded-md overflow-hidden">
+            <table className="w-full">
+              <thead className="bg-base-tertiary">
+                <tr>
+                  <th className="text-left p-3 text-sm font-medium">
+                    {t(I18nKey.SETTINGS$NAME)}
+                  </th>
+                  <th className="text-left p-3 text-sm font-medium">
+                    {t(I18nKey.SETTINGS$KEY_PREFIX)}
```

Comment:
Is it worth adding the key prefix field? Otherwise is it possible to view the prefix, it seems hidden by default without a way of viewing it

Reply:
EDIT: actually it looks like the prefix field isn't being populated, so I just see triple dots; probably requires a change on deploy repo

Reply:
Yeah you're right, something is up here. Can fix in the future

@openhands remove the prefix column from the table